diff --git a/SuperBuild/External_python.cmake b/SuperBuild/External_python.cmake
index 2216b9570a..166daeafc3 100644
--- a/SuperBuild/External_python.cmake
+++ b/SuperBuild/External_python.cmake
@@ -54,9 +54,21 @@ if((NOT DEFINED PYTHON_INCLUDE_DIR
   set(_download_3.9.10_url "https://www.python.org/ftp/python/3.9.10/Python-3.9.10.tgz")
   set(_download_3.9.10_md5 "1440acb71471e2394befdb30b1a958d1")
 
+  ExternalProject_SetIfNotDefined(
+    Slicer_${proj}_ARCHIVE_URL
+    "${_download_3.9.10_url}"
+    QUIET
+  )
+
+  ExternalProject_SetIfNotDefined(
+    Slicer_${proj}_ARCHIVE_MD5
+    "${_download_3.9.10_md5}"
+    QUIET
+  )
+
   ExternalProject_Add(python-source
-    URL ${_download_${Slicer_REQUIRED_PYTHON_VERSION}_url}
-    URL_MD5 ${_download_${Slicer_REQUIRED_PYTHON_VERSION}_md5}
+    URL ${Slicer_${proj}_ARCHIVE_URL}
+    URL_MD5 ${Slicer_${proj}_ARCHIVE_MD5}
     DOWNLOAD_DIR ${CMAKE_BINARY_DIR}
     SOURCE_DIR ${python_SOURCE_DIR}
     CONFIGURE_COMMAND ""
