From 4232523653c5731b3eeaae00b26f21d98fdefa35 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Fri, 5 May 2023 07:07:27 +0200
Subject: [PATCH 11/14] ENH: Enable build with offilne PCRE

---
 SuperBuild/External_PCRE.cmake | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/SuperBuild/External_PCRE.cmake b/SuperBuild/External_PCRE.cmake
index 33bd31c861..d53a631841 100644
--- a/SuperBuild/External_PCRE.cmake
+++ b/SuperBuild/External_PCRE.cmake
@@ -49,10 +49,22 @@ ExternalProject_Execute(${proj} \"configure\" sh ${EP_SOURCE_DIR}/configure
 
   set(_version "8.44")
 
+  ExternalProject_SetIfNotDefined(
+    Slicer_${proj}_ARCHIVE_URL
+    https://github.com/Slicer/SlicerBinaryDependencies/releases/download/PCRE/pcre-${_version}.tar.gz
+    QUIET
+  )
+
+  ExternalProject_SetIfNotDefined(
+    Slicer_${proj}_ARCHIVE_SHA512
+    abac4c4f9df9e61d7d7761a9c50843882611752e1df0842a54318f358c28f5953025eba2d78997d21ee690756b56cc9f1c04a5ed591dd60654cc78ba16d9ecfb
+    QUIET
+  )
+
   ExternalProject_add(PCRE
     ${${proj}_EP_ARGS}
-    URL https://github.com/Slicer/SlicerBinaryDependencies/releases/download/PCRE/pcre-${_version}.tar.gz
-    URL_HASH SHA512=abac4c4f9df9e61d7d7761a9c50843882611752e1df0842a54318f358c28f5953025eba2d78997d21ee690756b56cc9f1c04a5ed591dd60654cc78ba16d9ecfb
+    URL ${Slicer_${proj}_ARCHIVE_URL}
+    URL_HASH SHA512=${Slicer_${proj}_ARCHIVE_SHA512}
     DOWNLOAD_DIR ${CMAKE_BINARY_DIR}
     SOURCE_DIR ${EP_SOURCE_DIR}
     BINARY_DIR ${EP_BINARY_DIR}
-- 
2.39.3

